src/solver/hydro/monthly/h2o_m_fonctions.h

```diff
@@ -27,15 +27,15 @@
 #ifndef __SOLVER_H2O_M_FONCTIONS__
 #define __SOLVER_H2O_M_FONCTIONS__
 
-DONNEES_ANNUELLES* H2O_M_Instanciation(int);
-void H2O_M_OptimiserUneAnnee(DONNEES_ANNUELLES*, int);
-void H2O_M_Free(DONNEES_ANNUELLES*);
+DONNEES_ANNUELLES H2O_M_Instanciation(int);
+void H2O_M_OptimiserUneAnnee(DONNEES_ANNUELLES&, int);
+void H2O_M_Free(DONNEES_ANNUELLES&);
 
-void H2O_M_ConstruireLesContraintes(DONNEES_ANNUELLES*);
-void H2O_M_ConstruireLesVariables(DONNEES_ANNUELLES*);
-void H2O_M_InitialiserBornesEtCoutsDesVariables(DONNEES_ANNUELLES*);
-void H2O_M_InitialiserLeSecondMembre(DONNEES_ANNUELLES*);
-void H2O_M_ResoudreLeProblemeLineaire(DONNEES_ANNUELLES*, int);
-void H2O_M_AjouterBruitAuCout(DONNEES_ANNUELLES*);
+void H2O_M_ConstruireLesContraintes(DONNEES_ANNUELLES&);
+void H2O_M_ConstruireLesVariables(DONNEES_ANNUELLES&);
+void H2O_M_InitialiserBornesEtCoutsDesVariables(DONNEES_ANNUELLES&);
+void H2O_M_InitialiserLeSecondMembre(DONNEES_ANNUELLES&);
+void H2O_M_ResoudreLeProblemeLineaire(DONNEES_ANNUELLES&, int);
+void H2O_M_AjouterBruitAuCout(DONNEES_ANNUELLES&);
```

Comment:
Make a C++ class with these methods ? Most of them private, probably just a few public.

Instantiation => constructor
Free => destructor